src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -3732,48 +3742,139 @@ public enum DarknessAblaze implements LogicCardInfo {
       return itemCard (this) {
         text "Search your deck for a Pokémon with the same name as a Pokémon in your discard pile, and add it to your hand. Then, shuffle your deck."
         onPlay {
+          def pokeNamesInDiscard = []
+          my.discard.filterByType(POKEMON).each{
+            pokeNamesInDiscard.add(it.name)
+          }
+          my.deck.search("Pokémon with same name as a Pokémon in your discard pile. Possible options: $pokeNamesInDiscard", {it.cardTypes.is(POKEMON) && pokeNamesInDiscard.contains(it.name)}).moveTo(my.hand)
         }
         playRequirement{
+          my.deck : "You have no cards in your deck"
+          my.discard.filterByType(POKEMON) : "You have no Pokémon in your discard pile"
         }
       };
       case PIERS_173:
       return supporter (this) {
-        text "Search your deck for 1 [D] Pokémon and any 1 Energy, reveal them, and put them into your hand. Then, shuffle your deck."
-        onPlay {
-        }
-        playRequirement{
-        }
+        text "Search your deck for a [D] Pokémon and an Energy card, reveal them, and put them into your hand. Then, shuffle your deck."
+          onPlay {
+            my.deck.search(count: 1, "Search your deck for a [D] Pokémon",pokemonTypeFilter(D)).moveTo(my.hand)
+            my.deck.search("Search your deck for an Energy card",cardTypeFilter(ENERGY)).moveTo(my.hand)
+            shuffleDeck()
+          }
+          playRequirement{
+            assert my.deck.notEmpty
+          }
       };
       case POKEMON_BREEDER_S_NURTURING_174:
       return supporter (this) {
         text "Choose up to 2 of your Pokémon in play. Search your deck for a card that evolves from each of those Pokémon and put it on top of the Pokémon to evolve it. Then, shuffle your deck. (You can’t use this card during your first turn or on a Pokémon that just came into play.)"
         onPlay {
+          def pcsList = my.all.findAll {it.turnCount < bg.turnCount}
+          if (pcsList.size() > 1) {
+            pcsList = pcsList.select(max:2, "Choose up to 2 of your Pokémon in play that can make a regular evolve during this turn.")
+          }
+
+          pcsList.each{ pcs =>
+            def sel = deck.search ("Select a Pokémon that evolves from $pcs.name.", {
+              it.cardTypes.is(EVOLUTION) && it.predecessor == pcs.name
+            })
+            if (sel) { evolve(pcs, sel.first(), OTHER) }
+          }
+
+          shuffleDeck()
         }
         playRequirement{
+          assert my.deck : "Your deck is empty."
+          assert bg.turnCount > 2 : "Cannot use this card during your first turn."
+          assert my.all.findAll {it.turnCount < bg.turnCount}
```

Comment:
You also need to check that there is at least a possibility of an evolution. I'm not sure how to do that currently, but the relevant ruling I do have

> Q. Can I use Wally or Evosoda on a Snorlax to search my deck, even though there is {currently} no evolution of Snorlax?
A. No, there has to at least be the possibility of a valid evolution in order to use this card. Sorry. (Jul 26, 2018 TPCi Rules Team)

Reply:
Hmm, I agree on this. But it'd probably need a special method on the server, similar to "basicsFromStage2" does for cards like rare candy.

Said method though, should not check for specific formats:

> Q. Can I use Wally or Evosoda to search my deck for a card that is not in the current tournament format? For example, could I use it on Raichu to search for a Raichu BREAK after it rotates out?
>
> A. Yes, you can. Even though a card may not be in the tournament format it still exists and can be searched for even though it is known to fail. (Jul 26, 2018 TPCi Rules Team)

Reply:
(cc @axpendix)

Reply:
that is doable but tough, there is currently no way to expose stage1 -> stage2 and basic -> stage1 evolution lines. the system that delivers basic -> stage2 evolution lines is called `carddb-api` and it is a separate module of its own written in spring boot. if you are interested in adding that extra endpoint, I am willing to do the engine integration to your endpoint which will let us implement this card perfectly. maybe that could also be used in some other similar type of cards such as evolution fields and such. @flagrama @starg09

Reply:
I'm certainly interested in doing anything that can help us make the cards more accurate when played. Might need pointed out where the basic -> stage2 thing is being done, but hopefully can figure it out from there.

Reply:
sent you the invite @flagrama 
contact me on discord